code/addons/controls/template/stories/basics.stories.ts

```diff
@@ -4,7 +4,8 @@ import type { PartialStoryFn, StoryContext } from '@storybook/types';
 export default {
   component: globalThis.Components.Pre,
   decorators: [
-    (storyFn: PartialStoryFn, context: StoryContext) => storyFn({ args: { object: context.args } }),
+    (storyFn: PartialStoryFn, context: StoryContext) =>
+      storyFn({ args: { object: { ...context.args } } }),
```

Comment:
So the problem is assigning `args.object = args` blows up stringify? I guess what we are doing in these decorators is pretty bizarre so having to do hacks is fine.

Reply:
That is right, it blows up stringify! 
Okay, so then I just do it like this.

Reply:
@kasperpeulen @tmeasday  it is very weird to impose spreading args ```storyFn({ args: { object: { ...context.args } } })``` in order for it to work,

Reply:
@chakAs3 what we are doing in these stories (assigning the whole args object into a single arg) is very weird, so I wouldn't worry about it (from a user perspective).

Reply:
Yes i know i dealt with it when I was rewriting vue renderer, it was weird for me the first time but after that I found it legit and even useful, to alter the story's args,   using the decorator that can cherry pick some args and passe them to single object prop for the component should work 
```ts 
  // Compose the set of  args into `object`, so the pre component only needs a single prop
  //   (selecting only the args specified on parameters.argNames if set)
  decorators: [
    (storyFn: PartialStoryFn, context: StoryContext) => {
      const { argNames } = context.parameters;
      const object = argNames ? pick(context.args, argNames) : context.args;
      return storyFn({ args: { object } });
    },
  ],
  ```

Reply:
@tmeasday   check https://stackblitz.com/~/github.com/storybookjs/vue3-code-examples,
I have included all these cases and it just working fine.  you don't have to spread your context.args

Reply:
@chakAs3 has something changed to make it no longer necessary? @kasperpeulen added the spreading because this change mean the args object became self-referential in these stories and thus created problems when it was serialized over the channel. At the time I played with it a bit and saw the problem too.

<img width="1555" alt="image" src="https://github.com/storybookjs/storybook/assets/132554/72d27303-809d-455f-82f0-23bad042d9c0">